src/sentry/services/hybrid_cloud/user/__init__.py

```diff
@@ -64,6 +65,9 @@ def get_avatar_type(self) -> str:
     def class_name(self) -> str:
         return "User"
 
+    def has_2fa(self) -> bool:
+        return self.authenticators is not None and len(self.authenticators) > 0
```

Comment:
Should this method raise if the user wasn't hydrated with authenticators? Currently this looks like it could cloak mistakes like that.

Reply:
I agree, and I'll add.